python/src/main.cpp

```diff
@@ -2,13 +2,25 @@
 #include <pybind11/stl.h>
 #include <pybind11/numpy.h>
 
-#include <kompute/Kompute.hpp>
+#include <kompute/Core.hpp>
```

Comment:
So the single header kompute is the main issue? It may be best to try to see if it can be remediated - would perhaps updating the order of the includes in the Aggregate Headers file help? https://github.com/EthicalML/vulkan-kompute/blob/master/single_include/AggregateHeaders.cpp

Reply:
Will try, but I think it's better to directly include the required headers for easier development when an interface changes.

Reply:
Perfect, thanks @alexander-g !

Yeah I do agree, including only what is required would definitely be better once users get up to speed - although potentially this may have uncovered an issue that could come up if users include the dependencies in the same order as the AggregateHeaders

Reply:
Turns out rebuilding the single header (without re-ordering) already helped. No idea why. Nevertheless, I have left the individual includes as mentioned above

Reply:
Makes sense as it's still an "internal" package, so we can include the files as separate - ok so perhaps it could be due to it building on windows vs linux (the file in the repo is built via windows, hence windows style newlines). Strange, then it's good to know in case other people are having issues - although it's not really clear what specifically is causing the issue, as I also see the logger bug when running it in Windows

Reply:
Actually just looking at the AggregateHeaders file, it has almost the same imports, so it may be best to include it directly, as you will pretty much be importing the same files (https://github.com/EthicalML/vulkan-kompute/blob/master/single_include/AggregateHeaders.cpp). 

The main thinking here is that it would make sense to import only a subset of files when not everything is needed, but given the Python dependency imports everything only once, it makes sense to just have the single Kompute.cpp.

Just to confirm, so you managed to get it to work? Did you see any particular diffs between the previous one and the new one by any chance?

Reply:
Ok, I've replaced the single includes with `AggregateHeaders`.
Yes, seems to work, diffs did not show anything special.

Reply:
So strange! Thank you for having a look - do you think there could be a python test we could add to check this by any chance?

Reply:
To be honest I can't really think how, unless we create a log_level getter, I guess that way at least it could check if the state of spdlog loglevel has been changed?

Reply:
Added #120 so this can be tackled on a separate PR